src/conv/heuristics/ai_heuristics.cpp

```diff
@@ -500,87 +475,346 @@ class Gfx942Model final : public Model
+        auto cached_result = GetCachedPrediction(problem, device, false); // false = 2D
+        if(!cached_result.empty()) {
+            return cached_result;
+        }
+        
+        // Only create model if cache miss - expensive but necessary
+        std::unique_ptr<Model> model = GetModel(device);
+        if(!model || !model->IsProblemSupported(problem, ctx))
+            return {};
+
+        MIOPEN_LOG_I2("Evaluating TunaNet");
+        std::vector<float> res = model->Forward(problem);
+
+        // Process predictions using helper function
+        auto result = ProcessPredictions(res, model->metadata.solver_map, false); // false = 2D
+        
+        StorePredictionCache(problem, device, false, result.any_solver_ids); // false = 2D
+        if(miopen::IsLogging(LoggingLevel::Info2))
+        {
+            std::stringstream ss;
+            for(auto& id : result.solver_ids)
+                ss << solver::Id{id}.ToString() << " ID:" << id << ", ";
+            MIOPEN_LOG_I2("TunaNet Result: " << ss.str());
+        }
+        return result.solver_ids;
     }
-    return sol;
 }
+
 } // namespace immed_mode
+
+/**
+ * @brief 3D convolution AI heuristics namespace
+ * 
+ * This namespace contains classes and functions for 3D convolution AI heuristics
+ * using TunaNet3D neural networks to predict optimal solvers for 3D convolution
+ * operations (NCDHW layout).
+ */
+namespace conv3d {
+
+// Metadata3D implementation moved to metadata_3d.cpp
+
+class Gfx942Model_3D : public Model3D
+{
+private:
+    const std::string arch_name;
+public:
+    Metadata3D metadata;
+    
+    Gfx942Model_3D() : arch_name("gfx942_3d"), metadata(Metadata3D(arch_name)) {
+        MIOPEN_LOG_I2("Gfx942Model_3D initialized");
+    }
+    
+    std::vector<float> Forward(const conv::ProblemDescription& problem) const override
+    {
+        std::vector<float> features = ToFeatures(problem);
+        MIOPEN_LOG_I2("Gfx942Model_3D: Extracted " << features.size() << " features");
+        
+        // Use fdeep to run TunaNet3D inference
+        const std::string model_path = Model3DPath(arch_name);
+        const auto model = fdeep::load_model(model_path);
+        
+        // Convert features to fdeep tensor
+        const auto input_tensor = fdeep::tensor(fdeep::tensor_shape(features.size()), features);
+        const auto result = model.predict({input_tensor});
+        
+        // Extract predictions from result
+        const auto predictions = result[0].to_vector();
+        MIOPEN_LOG_I2("Gfx942Model_3D: TunaNet3D returned " << predictions.size() << " predictions");
+        return predictions;
+    }
+    
+    const std::unordered_map<size_t, std::string>& GetSolverMap() const override
+    {
+        return metadata.GetSolverMap();
+    }
+    
+    bool IsProblemSupported(const conv::ProblemDescription& problem,
+                            const ExecutionContext& ctx) const override
```

Comment:
I receive a warning when building, since ctx is not used, perhaps do:

```suggestion
                            [[maybe_unused]] const ExecutionContext& ctx) const override
```

Reply:
It is easier to simply write 
```
bool IsProblemSupported(const conv::ProblemDescription& problem,
                            const ExecutionContext&) const override
```

If you are not using the `ctx`. This should remove the compiler warning.

Reply:
Thank you Bart and Ville, I corrected.